pkg/internal/layers/layers.go

```diff
@@ -85,11 +69,33 @@ type Layer interface {
 	GetAddonsLayers() (map[string]*kraanv1alpha1.AddonsLayer, error)
 }
 
+// KraanLayer is the Schema for the addons API.
+type KraanLayer struct {
+	updated     bool
+	requeue     bool
```

Comment:
I really think that `requeue, delayed, delay` shouldn't be needed as the reconcile flow could determine that